apps/dashboard/src/api/api.client.ts

```diff
@@ -0,0 +1,70 @@
+import { getToken } from '@/utils/auth';
+import { API_ROOT } from '../config';
+import { getEnvironmentId } from '@/utils/local-storage';
+
+class NovuApiError extends Error {
+  constructor(
+    message: string,
+    public error: unknown,
+    public status: number
+  ) {
+    super(message);
+  }
+}
+
+type HttpMethod = 'GET' | 'POST' | 'PUT' | 'DELETE';
+
+const request = async <T>(
+  endpoint: string,
+  method: HttpMethod = 'GET',
+  data?: unknown,
+  headers?: HeadersInit
+): Promise<T> => {
+  try {
+    const jwt = await getToken();
+    const environmentId = getEnvironmentId();
+    const config: RequestInit = {
+      method,
+      headers: {
+        Authorization: `Bearer ${jwt}`,
+        'Content-Type': 'application/json',
+        ...(environmentId && { 'Novu-Environment-Id': environmentId }),
+        ...headers,
+      },
+    };
+
+    if (data) {
+      config.body = JSON.stringify(data);
+    }
+
+    const baseUrl = API_ROOT ?? 'https://api.novu.co';
+    const response = await fetch(`${baseUrl}/v1${endpoint}`, config);
+
+    if (!response.ok) {
+      const errorData = await response.json();
+      throw new NovuApiError(`Novu API error`, errorData, response.status);
+    }
+
+    if (response.status === 204) {
+      return {} as T;
+    }
+
+    return await response.json();
```

Comment:
If not mistaken we need another `type assertion here`

Reply:
not needed because `response.json()` type is `any` which is assigned as `T`